File: src/MixNetFx/FacFunctions/Dialogs/CostcenterInfoRelated/frmCostCenterSelector.vb
Imports Facesso.Data
Imports System.Windows.Forms

Public Class frmCostCenterSelector

    Public Function HandleDialog(ByVal ccic As CostcenterInfoItems) As CostcenterInfo
        Using Me
            lstCostCenter.DataSource = ccic
            lstCostCenter.DisplayMember = "ListItemText"
            Me.ShowDialog()
            If DialogResult = Windows.Forms.DialogResult.OK Then
                Return DirectCast(lstCostCenter.SelectedItem, CostcenterInfo)
            Else
                Return Nothing
            End If
        End Using
    End Function

    Private Sub btnOK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOK.Click
        If lstCostCenter.SelectedIndex = -1 Then
            MessageBox.Show(My.Resources.CostCenterInfoCenter_NoSelectedCostCenter_MB_Body, _
                            My.Resources.CostCenterInfoCenter_NoSelectedCostCenter_MB_Title, _
                             MessageBoxButtons.OK, MessageBoxIcon.Exclamation)
            Exit Sub
        End If
        Me.DialogResult = Windows.Forms.DialogResult.OK
    End Sub

    Private Sub btnCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancel.Click
        Me.DialogResult = Windows.Forms.DialogResult.Cancel
    End Sub
End Class

File: src/NetFxCSharp/Facesso.Functions/Dialogs/CostcenterInfoRelated/frmCostCenterSelector.cs
using Facesso.Data;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Xml.Linq;

namespace Facesso.Functions
{
    public partial class frmCostCenterSelector
    {
        public CostcenterInfo HandleDialog(CostcenterInfoItems ccic)
        {
            using (this)
            {
                lstCostCenter.DataSource = ccic;
                lstCostCenter.DisplayMember = "ListItemText";
                this.ShowDialog();
                if (DialogResult == System.Windows.Forms.DialogResult.OK)
                {
                    return ((CostcenterInfo)lstCostCenter.SelectedItem);
                }
                else
                {
                    return null;
                }
            }

            return default(CostcenterInfo);
        }

        private void btnOK_Click(System.Object sender, System.EventArgs e)
        {
            if (lstCostCenter.SelectedIndex == -1)
            {
                MessageBox.Show(Facesso.Functions.My.Resources.CostCenterInfoCenter_NoSelectedCostCenter_MB_Body, Facesso.Functions.My.Resources.CostCenterInfoCenter_NoSelectedCostCenter_MB_Title, MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
                return;
            }

            this.DialogResult = System.Windows.Forms.DialogResult.OK;
        }

        private void btnCancel_Click(System.Object sender, System.EventArgs e)
        {
            this.DialogResult = System.Windows.Forms.DialogResult.Cancel;
        }

        public frmCostCenterSelector()
        {
            InitializeComponent();
        }
    }
}